ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -619,4 +621,14 @@ default void prepareAlterTableEnvironmentContext(AbstractAlterTableDesc alterTab
   default Boolean hasAppendsOnly(org.apache.hadoop.hive.ql.metadata.Table hmsTable, SnapshotContext since) {
     return null;
   }
+
+  default List<String> showPartitions(DDLOperationContext context,
+      org.apache.hadoop.hive.ql.metadata.Table tbl) throws UnsupportedOperationException {
+    throw new UnsupportedOperationException("Storage handler does not support show partitions command");
+  }
+
+  default boolean supportsPartitions(){
```

Comment:
why do we need this? it's ok to call `default` showPartitions that might return `UnsupportedOperationException`

Reply:
Fixed